template.yaml

```diff
@@ -243,6 +264,22 @@ Resources:
       SourceArn: !GetAtt PersistedResourcesEventRule.Arn
 
   #============================= Roles ==========================================================
+
+  DataWarehouseRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: "arn:aws:iam::${DataWarehouseAccountId}:root"
+            Action: sts:AssumeRole
+            Condition:
+              StringEquals:
+                sts:ExternalId: !Ref DataWarehouseExternalId
```

Comment:
Can you explain what this does?

Reply:
It's an extra security measure, when someone from the other aws account assumes this role via the aws Security Token Service API (sts), they must provide an external id.

See linked docs:  https://aws.amazon.com/blogs/apn/securely-using-external-id-for-accessing-aws-accounts-owned-by-others/

Reply:
It's to prevent other accounts from assuming this role via data warehouse account.

Reply:
Can we add a description to the role to say that this provides external access when paired with known external account and Id?